### PR TITLE
RavenDB-14205 - Restore from backup not respecting "Disable ongoing tasks after restore"

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14205

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs [hunk 1]
```diff
@@ -598,6 +598,30 @@ private void DisableOngoingTasksIfNeeded(DatabaseRecord databaseRecord)
                     task.Disabled = true;
                 }
             }
+
+            if (databaseRecord.ExternalReplications != null)
+            {
+                foreach (var task in databaseRecord.ExternalReplications)
+                {
+                    task.Disabled = true;
```
Comment (ayende) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:606]: Shouldn't this come from the restore config?

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs [hunk 2]
```diff
@@ -598,6 +598,30 @@ private void DisableOngoingTasksIfNeeded(DatabaseRecord databaseRecord)
...
+                }
+            }
+
+            if (databaseRecord.HubPullReplications != null)
+            {
+                foreach (var task in databaseRecord.HubPullReplications)
+                {
+                    task.Disabled = true;
```
Comment (ayende) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:614]: What if  the value was already disabled at backup time?

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs [hunk 3]
```diff
@@ -260,6 +258,8 @@ public async Task<IOperationResult> Execute(Action<IOperationProgress> onProgres
                                 await SmugglerRestore(database, filesToRestore, context, databaseRecord, onProgress, result);
                             }
 
+                            DisableOngoingTasksIfNeeded(databaseRecord);
```
Comment (ayende) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:261]: why move it here? The tasks are enabled at this point, no?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:261]: This is after we restored the database record but before we create the database:
https://github.com/ravendb/ravendb/blob/462b73c58d4d9d7f48081dc305e053b9cbed3efa/src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs#L278

Comment (ayende) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:261]: ok

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:606]: We skip it if it was enabled in the restore config:
https://github.com/ravendb/ravendb/blob/462b73c58d4d9d7f48081dc305e053b9cbed3efa/src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs#L567

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:614]: It will stay disabled

**Review by ayende — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs [hunk 4]
```diff
@@ -107,6 +107,7 @@ public virtual DynamicJsonValue ToJson()
             var result = new DynamicJsonValue
             {
                 [nameof(Name)] = Name,
+                [nameof(TaskId)] = TaskId,
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs:110]: Why do we need this?

Comment (arekpalinski) [reply src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs:110]: I'm fine with that, just wonder why did you need to add it?

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs (hunk 4 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs:110]: It fails here after restore:
https://github.com/ravendb/ravendb/blob/462b73c58d4d9d7f48081dc305e053b9cbed3efa/src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs#L79

Comment (arekpalinski) [reply src/Raven.Client/Documents/Operations/ETL/EtlConfiguration.cs:110]: Makes sense. After restore `TaskId` should be the same.

**Pull request merged**